### ISSUE TITLE
dashboard redirect loose scheme

### ISSUE THREAD

**Issue by frntn** [neo4j/neo4j#4268]
### Context

I have a neo4j dashboard (7474) behind a reverse proxy binded on https (443) 

### What I get

When trying to access the dashboard from https I get redirected to http

```bash
curl -ILk 'https://neo4j-redirect-issue/' -H 'Accept: text/html' 
HTTP/1.1 303 See Other
Location: https://neo4j-redirect-issue/browser
Content-Type: text/html
Access-Control-Allow-Origin: *
Server: Jetty(9.0.5.v20130815)

HTTP/1.1 302 Found
Location: http://neo4j-redirect-issue/browser/
Server: Jetty(9.0.5.v20130815)

curl: (7) Failed to connect to neo4j-redirect-issue port 80: Connection refused
```

Here we see the scheme is kept in the first 303 redirect, but lost in the next 302 (when adding the trailing slash).

### What I expect

I expect the scheme to be kept along all redirects so a call to `https://neo4j-redirect-issue/` end up to `https://neo4j-redirect-issue/browser/`

**Comment by cleishm** [comment]
Duplicate of #1488. Resolved by #4244 and #4246 (not yet released, will be in 2.1.8 and 2.2.0).

**Comment by cleishm** [comment]
BTW: when using a reverse proxy, you should have the proxy set the `X-FORWARDED-*` headers as described in the documentation (http://neo4j.com/docs/stable/server-configuration.html#_using_x_forwarded_proto_and_x_forwarded_host_to_parameterize_the_base_uri_for_rest_responses)
